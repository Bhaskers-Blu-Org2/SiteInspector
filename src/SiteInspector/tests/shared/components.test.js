// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.

import React from 'react';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Blade from '../../shared/components/Blade';
import Tab from '../../shared/components/Tab';
import TabNavigation from '../../shared/components/TabNavigation';

configure({ adapter: new Adapter() });

describe('Shared components', () => {
  describe('Blade component', () => {
    let blade;

    beforeEach(() => {
      blade = shallow(<Blade
        title="test title"
        closeHandler={jest.fn()}
        displayUI={<div>hello world</div>}
        isLargerWidth={false}
      />);
    });

    it('should be created', () => {
      expect(blade.length).toEqual(1);
    });

    it('should be rendered properly', () => {
      expect(blade).toMatchSnapshot();
    });
  });

  describe('Tab component', () => {
    let tab;
    const tabData = {
      id: 'test',
      label: 'test label',
    };

    beforeEach(() => {
      tab = shallow(<Tab
        tab={tabData}
        setViewer={jest.fn()}
      />);
    });

    it('should be created', () => {
      expect(tab.length).toEqual(1);
    });

    it('should be rendered properly', () => {
      expect(tab).toMatchSnapshot();
    });
  });

  describe('TabNavigation component', () => {
    let tabNav;
    const tabData = {
      id: 'test',
      label: 'test label',
      view: <div>test</div>,
    };

    beforeEach(() => {
      tabNav = shallow(<TabNavigation
        tabs={{ test: tabData }}
        activeTab="test"
        setTab={jest.fn()}
      />);
    });

    it('should be created', () => {
      expect(tabNav.length).toEqual(1);
    });

    it('should be rendered properly', () => {
      expect(tabNav).toMatchSnapshot();
    });
  });
});
