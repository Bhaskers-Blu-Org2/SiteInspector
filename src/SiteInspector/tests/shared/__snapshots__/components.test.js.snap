// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Shared components Blade component should be rendered properly 1`] = `
<div
  className="si-blade flex-column "
>
  <div
    className="si-blade-header"
  >
    <div
      className="si-blade-header-title"
    >
      test title
    </div>
    <a
      className="si-close-panel-icon c-glyph glyph-clear"
      href="javascript:void(0)"
      onClick={[MockFunction]}
    />
  </div>
  <div
    className="si-blade-content flex-column"
  />
</div>
`;

exports[`Shared components Tab component should be rendered properly 1`] = `
<a
  className="si-tab c-select-button "
  href="javascript:void(0)"
  onClick={[Function]}
>
  test label
</a>
`;

exports[`Shared components TabNavigation component should be rendered properly 1`] = `
<div
  className="si-tab-navigation flex-column"
>
  <div
    className="si-nav"
  >
    <div
      className="c-group"
      role="tablist"
    >
      <SITab
        isActive={true}
        key="test"
        setViewer={[MockFunction]}
        tab={
          Object {
            "id": "test",
            "label": "test label",
            "view": <div>
              test
            </div>,
          }
        }
      />
    </div>
  </div>
  <div
    className="si-tab-content flex-column"
  >
    <div>
      test
    </div>
  </div>
</div>
`;
